pkg/apis/serving/v1/revision_defaults.go

```diff
@@ -51,52 +52,78 @@ func (rs *RevisionSpec) SetDefaults(ctx context.Context) {
 
 	for idx := range rs.PodSpec.Containers {
 		if rs.PodSpec.Containers[idx].Name == "" {
-			rs.PodSpec.Containers[idx].Name = cfg.Defaults.UserContainerName(ctx)
-		}
-
-		if rs.PodSpec.Containers[idx].Resources.Requests == nil {
-			rs.PodSpec.Containers[idx].Resources.Requests = corev1.ResourceList{}
-		}
-		if _, ok := rs.PodSpec.Containers[idx].Resources.Requests[corev1.ResourceCPU]; !ok {
-			if rsrc := cfg.Defaults.RevisionCPURequest; rsrc != nil {
-				rs.PodSpec.Containers[idx].Resources.Requests[corev1.ResourceCPU] = *rsrc
-			}
-		}
-		if _, ok := rs.PodSpec.Containers[idx].Resources.Requests[corev1.ResourceMemory]; !ok {
-			if rsrc := cfg.Defaults.RevisionMemoryRequest; rsrc != nil {
-				rs.PodSpec.Containers[idx].Resources.Requests[corev1.ResourceMemory] = *rsrc
+			if len(rs.PodSpec.Containers) > 1 {
+				rs.PodSpec.Containers[idx].Name = cfg.Defaults.UserContainerName(ctx) + "-" + randString(5, "abdpq")
```

Comment:
why not just generate a UUID?
Also, strictly theoretically, there is probably a limit on the length of this string, so if I update the UCN to be something long it might cause problems?

Reply:
Do you mean this UUID https://godoc.org/github.com/google/uuid#UUID.String ?

Reply:
For example, or some variations thereof. Do you know if there's a length limit for the container name? 63?

Reply:
Yes there is a length Limit for name